### DISCUSSION TITLE
Merge, Rebase dan Squash?

### DISCUSSION THREAD

**Discussion by urvangibran** [Lab-RPL-ITS/module-pelatihan-vcs#14] [Q&A] [answered]
Perbedaan merge sama reabase tuh apa ya mas? terus penggunaan yang tepat buat merge itu waktu apa dan reabase waktu apa

**Comment by dzikrisyairozi** [comment, marked as answer]
Perbedaan penggunaan lainnya itu biasanya ada di kompleksitas proyeknya. Kalau di project scale open source yang dimana setiap orang (selain repo maintainer) ngga bisa buat branch di dalam repo itu sendiri, biasanya untuk mengajukan perubahan harus fork repo tersebut terlebih dahulu, saat pull request via fork dilakukan setiap perubahannya akan dilakukan squash agar commit terlihat rapi. 
Berbeda dengan private project yang lebih kecil, dimana pengerjaan setiap fitur itu dilakukan pada branch masing masing, merge atau rebase commit sudah cukup tergantung preferensi tim.
